### PR TITLE
Added Sum of Digits Implementation

### PR DESCRIPTION
**Pull request by SpiderMath**
The related question: https://the-algorithms.com/algorithm/sum-of-digits

There's no JavaScript implementation, so I implemented in JS. Also, I might have messed up writing the tests, since it was not exactly clear to me. So I'd request to please crosscheck the tests(, and if I have any mistake/error in there, I'll fix it)

### CODE REVIEW

**Comment by SpiderMath** [comment]
Fixed the mistake

File: Maths/SumOfDigits.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	sumOfDigits(10) will return 1 + 0 = 1
+	sumOfDigits(255) will return 2 + 5 + 5 = 12
+	Wikipedia: https://en.wikipedia.org/wiki/Digit_sum
+*/
+
+/*
+	The given input is converted to a string, split into an array of characters.
+	This array is reversed and joined back together to form a string. which is finally converted to a number.
+*/
+function sumOfDigitsUsingString(number) {
```
Comment (raklaptudirm) [inline Maths/SumOfDigits.js:12]: How is this a valid implementation of the algorithm you propose? This will just reverse the number.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Maths/SumOfDigits.js (hunk 1 above)
Comment (SpiderMath) [reply Maths/SumOfDigits.js:12]: Oops, I'm really sorry. I messed up. I'll fix it right away.

Comment (SpiderMath) [reply Maths/SumOfDigits.js:13]: Fixed the implementation

I intended to write the application I did now, but I kinda messed them up. Extremely sorry for that.

**Review by raklaptudirm — APPROVED** [review]

**Comment by raklaptudirm** [comment]
What happened to the `package-lock.json`?

**Comment by SpiderMath** [comment]
Umm, I tried to install the Node Modules which were there for the project (you hadn't told me about Jest testing stuff on Discord yet) and I think the conflict might be because of that.

Also, I'll fix the 2 spaces problem, I thought setting tab space to 2 would suffice but looks like it didn't.

**Comment by raklaptudirm** [comment]
For the styling problem, you need to style the code with [`standard.js`](https://standardjs.com/).

**Comment by SpiderMath** [comment]
Okay finally. Sorry I have nearly no experience with linters (I just have a eslintrc which use everywhere) so I kinda messed up...

**Comment by raklaptudirm** [comment]
`standard.js` is easy to use with no configuration needed, so in my opinion it is the best for beginners, who are targeted by this repo.

**Comment by SpiderMath** [comment]
I can't help but agree. But as a guy mostly used to ESLint and one who didn't try anything else, it was confusing. 

[`Standard`](https://npmjs.com/package/standard) is really good, I don't doubt that. It was mostly my not-being used to it was what kinda messed them up. However, now that I know how it works, I will probably be able to contribute properly next time.

(Also can we merge the PR now? Looks like everything's fine now~)

**Comment by raklaptudirm** [comment]
Could you remove the `package-lock.json` changes?

**Comment by SpiderMath** [comment]
Umm, actually I don't exactly know how I could undo that :sweat_smile: 
Could you please tell me how to do that? ( Sorry if that sounds really dumb :cold_sweat: )

**Comment by SpiderMath** [comment]
Umm, but the thing which caused the changes in the `package-lock` was me running `npm install`
I still ran it again, and it generates the same Lockfile to the one I committed (by mistake).

Do you want me to copy the current `package-lock` from this Repo and put it into mine? Because that's the only other alternative I can think of at the moment.

**Comment by raklaptudirm** [comment]
If `npm install` keeps the file the same, leave it.

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**